src/core/plugin/luapi_application.h

```diff
@@ -1031,22 +1039,21 @@ static int applib_changeToolColor(lua_State* L) {
     }
 
     if (toolType == TOOL_NONE) {
-        g_warning("tool \"%s\" is not valid or no tool has been selected", toolTypeToString(toolType).c_str());
-        lua_pop(L, 3);
-        return 0;
+        // lua clears the stack when the function returns
```

Comment:
Is this comment true or should `lua_pop(L,3)` be restored?

Reply:
> From the point of view of C, a C function gets as its single argument the Lua state and returns (in C) an integer with the number of values it is returning (in Lua). Therefore, the function does not need to clear the stack before pushing its results. After it returns, Lua automatically removes whatever is in the stack below the results. 

According to the [PIL](https://www.lua.org/pil/26.1.html).
According to the [reference](https://www.lua.org/manual/5.4/manual.html#lua_CFunction) this behavior did not change (sadly always a possibility with lua).
Nevertheless, I decided to do this in an additional refactoring step as I stumbled over some functions where I'm not sure if the return value really is correct (and I as well want to finally complete this PR).

Thus, yes this comment is correct. Nonetheless `lua_pop(L, 3)` should be restored for now.